### PR TITLE
Let scheduled emails be edited & deleted.

### PR DESCRIPTION
**Pull request by AliengirlLiv**
Remember to run bin/delayed_job start beforehand so emails will be sent.

### CODE REVIEW

File: app/controllers/emails_controller.rb [hunk 1]
```diff
@@ -18,23 +23,96 @@ def create
...
+      end
+    end
+  end
 
-    # Run bin/delayed_job start to process all jobs
-    GeneralMailer.reminder_email_to_non_participants(subject, content, send_date)
+  def update
+    respond_to do |format|
+      email = params["email"]
+      params["email"]["sendDate"] = Date.new(email["sendDate(1i)"].to_i,
```
Comment (g-rocket) [inline app/controllers/emails_controller.rb]: If the sendDate of an email changes, do we need to set a new delayed_job to process it?

File: app/mailers/general_mailer.rb [hunk 2]
```diff
@@ -1,18 +1,53 @@
 class GeneralMailer < ApplicationMailer
-  def reminder_email_to_non_participants(subject, content, sendDate)
+  def self.reminder_email_to_non_participants(sendDate, emailId)
```
Comment (g-rocket) [inline app/mailers/general_mailer.rb]: Should we rename this function?

File: app/mailers/general_mailer.rb [hunk 3]
```diff
@@ -1,18 +1,53 @@
...
 
-  def reminder_email(user, subject, content)
+  def self.reminder_email(user, subject, content)
     @user = user
     mail(to: user.email, subject: subject, body: content)
   end
 
+  def scheduled_email(emailId)
+    email = Email.find emailId
+    if email
```
Comment (g-rocket) [inline app/mailers/general_mailer.rb]: Should we also check to see if the date has been updated?

File: app/views/emails/new.html.erb [hunk 4]
```diff
@@ -1,28 +1,39 @@
...
-         <h5 style='color: #E5E7E9'> <%= student.user.email%> </h5>
-      <% end %>
-    <% end %>
-<% end %>
-
-<br>
-
 <body class="title">
 
 <%= form_for @email, html: { multipart: true } do |f| %>
```
Comment (g-rocket) [inline app/views/emails/new.html.erb]: We can share the same form between create and edit; see how it's done for students.

File: app/views/emails/_editForm.html.erb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+    <%= f.text_field :subject, class: 'subject', :required => true, :placeholder => 'Subject'%>
+    <br>
+    <%= f.text_field :description, class: 'message', :required => true, :placeholder => 'Message'%>
+    <br>
+    <%= f.date_select :sendDate %>
+    <br>
+    <%= f.submit "Save Edited Email"%>
+  <% end %>
+</body>
```
Comment (g-rocket) [inline app/views/emails/_editForm.html.erb]: What if you want to change what groups an email's being sent to?

File: app/views/emails/_editForm.html.erb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        <% end %>
+        </ul>
+      </div>
+    <% end %>
+
+    <%= f.text_field :subject, class: 'subject', :required => true, :placeholder => 'Subject'%>
+    <br>
+    <%= f.text_field :description, class: 'message', :required => true, :placeholder => 'Message'%>
+    <br>
+    <%= f.date_select :sendDate %>
```
Comment (g-rocket) [inline app/views/emails/_editForm.html.erb]: Does changing the date work?

File: app/mailers/general_mailer.rb (hunk 2 above)
Comment (AliengirlLiv) [reply app/mailers/general_mailer.rb]: Fixed!

File: app/views/emails/new.html.erb (hunk 4 above)
Comment (AliengirlLiv) [reply app/views/emails/new.html.erb]: Fixed!

File: app/views/emails/_editForm.html.erb (hunk 5 above)
Comment (AliengirlLiv) [reply app/views/emails/_editForm.html.erb]: Fixed!

File: app/mailers/general_mailer.rb (hunk 3 above)
Comment (AliengirlLiv) [reply app/mailers/general_mailer.rb]: We now just delete emails when the send date changes.

File: app/controllers/emails_controller.rb (hunk 1 above)
Comment (AliengirlLiv) [reply app/controllers/emails_controller.rb]: Fixed!

File: app/views/emails/_editForm.html.erb (hunk 6 above)
Comment (AliengirlLiv) [reply app/views/emails/_editForm.html.erb]: Yep.

**Pull request merged**